rust_src/src/math.rs

```diff
@@ -291,3 +291,101 @@ fn abs(arg: LispObject) -> LispObject {
         }
     }
 }
+
+#[repr(C)]
+#[derive(Clone, Copy)]
+pub enum ArithComparison {
+    ArithEqual,
+    ArithNotequal,
+    ArithLess,
+    ArithGrtr,
+    ArithLessOrEqual,
+    ArithGrtrOrEqual,
+}
+
+#[no_mangle]
+pub extern "C" fn arithcompare(
+    num1: LispObject,
+    num2: LispObject,
+    comparison: ArithComparison,
+) -> LispObject {
+
+    let obj1 = check_number_coerce_marker(num1);
+    let obj2 = check_number_coerce_marker(num2);
+
+    let result = if obj1.is_float() || obj2.is_float() {
+        let f1 = obj1.any_to_float();
+        let f2 = obj2.any_to_float();
+
+        match comparison {
```

Comment:
This block can be extracted into a (inner) function generic over `T: PartialOrd+PartialEq`.